pkg/render/logstorage.go

```diff
@@ -522,22 +526,32 @@ func (es elasticsearchComponent) elasticsearchCluster() *esv1.Elasticsearch {
 			NodeSets: []esv1.NodeSet{
 				{
 					Count: int32(nodeConfig.Count),
-					Name:  "es",
+					Name: "es-" + nodeSetNameSuffix(pvcTemplate),
 					Config: &cmnv1.Config{
 						Data: map[string]interface{}{
 							"node.master": "true",
 							"node.data":   "true",
 							"node.ingest": "true",
 						},
 					},
-					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{es.pvcTemplate()},
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{pvcTemplate},
 					PodTemplate:          es.podTemplate(),
 				},
 			},
 		},
 	}
 }
 
+// nodeSetNameSuffix returns thumbprint of PersistentVolumeClaim object as string.
+// As storage requirements of NodeSets are immutable,
+// renaming a NodeSet automatically creates a new StatefulSet with new PersistentVolumeClaim.
+func nodeSetNameSuffix(pvcTemplate corev1.PersistentVolumeClaim) string{
+	pvcTemplateHash := md5.New()
+	templateBytes, _ := json.Marshal(pvcTemplate)
+	pvcTemplateHash.Write(templateBytes)
+	// Slice to fit NodeSets Name's max length of 23 char
```

Comment:
@Suraiya-Hameed could you reference the actual Elastic doc that states this? 

Important restriction and good to provide a reference here.

Reply:
I couldn't find a mention of this max length in the docs. Could you share the doc you found on this?

Reply:
https://github.com/elastic/cloud-on-k8s/blob/0404e804124f51fb7095f2ff2eb905b399067b42/pkg/apis/elasticsearch/v1beta1/elasticsearch_types.go#L63

Reply:
I couldn't find documentation around it, I got the error saying max length is 23 when I used full hash output (32 char).